I didn't have any hangs, but did find another bug (`super` needs new-style classes, so be it).  I also have some other failures in `iptest utils` which seem unrelated:
```
======================================================================
ERROR: Failure: ImportError (cannot import name json_clean)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/antony/venvs/ipy2/lib/python2.7/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/home/antony/venvs/ipy2/lib/python2.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/home/antony/venvs/ipy2/lib/python2.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/antony/venvs/ipy2/lib/python2.7/site-packages/IPython/utils/tests/test_jsonutil.py", line 14, in <module>
    from ..jsonutil import json_clean, encode_images
ImportError: cannot import name json_clean

======================================================================
ERROR: IPython.utils.tests.test_localinterfaces.test_load_ips
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/antony/venvs/ipy2/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/antony/venvs/ipy2/lib/python2.7/site-packages/IPython/utils/tests/test_localinterfaces.py", line 12, in test_load_ips
    localinterfaces._load_ips.called = False
AttributeError: 'module' object has no attribute '_load_ips'

======================================================================
ERROR: Failure: ImportError (cannot import name EventfulList)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/antony/venvs/ipy2/lib/python2.7/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/home/antony/venvs/ipy2/lib/python2.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/home/antony/venvs/ipy2/lib/python2.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/antony/venvs/ipy2/lib/python2.7/site-packages/IPython/utils/tests/test_traitlets.py", line 18, in <module>
    from IPython.utils.traitlets import (
ImportError: cannot import name EventfulList
======================================================================
FAIL: IPython.utils.tests.test_shimmodule.test_shim_warning
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/antony/venvs/ipy2/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/antony/venvs/ipy2/lib/python2.7/site-packages/IPython/utils/tests/test_shimmodule.py", line 14, in test_shim_warning
    assert len(w) == 1
AssertionError: 
>>  assert len([<warnings.WarningMessage object at 0x7fb13d2c9d90>, <warnings.WarningMessage object at 0x7fb13d2c9c50>, <warnings.WarningMessage object at 0x7fb13d2c9890>, <warnings.WarningMessage object at 0x7fb13d2f6fd0>]) == 1
    assert issubclass([<warnings.WarningMessage object at 0x7fb13d2c9d90>, <warnings.WarningMessage object at 0x7fb13d2c9c50>, <warnings.WarningMessage object at 0x7fb13d2c9890>, <warnings.WarningMessage object at 0x7fb13d2f6fd0>][-1].category, ShimWarning)
```

--- a/IPython/core/debugger.py
+++ b/IPython/core/debugger.py
@@ -206,7 +206,7 @@ def _file_lines(fname):
         return out
 
 
-class Pdb(OldPdb):
+class Pdb(OldPdb, object):
     """Modified Pdb class, does not load readline."""
 
     def __init__(self,color_scheme='NoColor',completekey=None,

